python/ml4ir/model/pipeline.py

```diff
@@ -244,15 +244,34 @@ def run(self):
                 ExecutionModeKey.TRAIN_INFERENCE,
                 ExecutionModeKey.INFERENCE_EVALUATE,
                 ExecutionModeKey.INFERENCE_ONLY,
+                ExecutionModeKey.INFERENCE_EVALUATE_RESAVE,
+                ExecutionModeKey.INFERENCE_RESAVE,
             }:
                 # Predict ranking scores
-                model.predict(
+                ranking_model.predict(
                     test_dataset=ranking_dataset.test,
                     inference_signature=self.args.inference_signature,
                     logs_dir=self.logs_dir,
                     logging_frequency=self.args.logging_frequency,
                 )
 
+            # Save model
+            # NOTE: Model will be saved with the latest serving signatures
+            if self.args.execution_mode in {
+                ExecutionModeKey.TRAIN_INFERENCE_EVALUATE,
```

Comment:
it seems like these should be broken down (in a separate story), into whether you want to save the model, train a model, etc.  Instead of one enum with a combinatorial explosion of different values.

Reply:
I agree.